subt/main.py

```diff
@@ -616,11 +626,24 @@ def stdout(self, *args, **kwargs):
         print(contents)
 
 #############################################
+    def system_nav_trace(self, goal):
+        """
+        Navigate along line
+        """
+        dx, dy, __ = self.offset
+        trace = Trace()
```

Comment:
Trace has implicit first point at [0,0,0]. In `update` the recording of the trace is going to be broken due to this.